cg/cli/demultiplex/demux.py

```diff
@@ -59,6 +59,7 @@ def demultiplex_all(context: CGConfig, flow_cells_directory: click.Path, dry_run
             continue
 
         if not dry_run:
+            demultiplex_api.remove_demultiplexing_output_directory(flow_cell)

```

Comment:
can we put this into the `demultiplex_api.start_demultiplexing(flow_cell=flow_cell)` function just before where we make the directory? 
Might even be worth considering to extract making and deleting the directory a separate function together. 
That function can check if exist, delete and create new.

Reply:
I'm would like to keep the deleting out of `start_demultiplexing` as it rather prepares before the start. But do you suggest extracting the directory creation out from `start_demultiplexing`?

Reply:
My suggestion would: 1. make the CLI cleaner. 2. Keep logic regarding the demux dir together. The idea was to move the deletion/creation/existence check closer together instread of having one in the CLI and the other calls in the start demultiplexing. Can be outside or inside the start demux call.

Reply:
I created the method `prepare_output_directory` which groups together delete and create methods. Put it outside of the start command. Could you take a quick look?

Reply:
Looks good to me"